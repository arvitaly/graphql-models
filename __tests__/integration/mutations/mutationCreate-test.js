"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const CreateDuplicateError_1 = require("./../../../CreateDuplicateError");
const utils_1 = require("./../utils");
const date1 = "Wed, 07 Dec 2016 10:42:48 GMT";
it("mutation create", () => __awaiter(this, void 0, void 0, function* () {
    const tester = new utils_1.default();
    const animals = [tester.generateModel("animal"), tester.generateModel("animal"), tester.generateModel("animal")];
    const post = tester.generateModel("post");
    const user = tester.generateModel("user");
    let current = 0;
    tester.adapter.findOne.mockImplementation((modelId, id) => {
        switch (modelId) {
            case "animal":
                return animals.find((r) => r.id === id);
            case "post":
                return Object.assign({}, post, {
                    animals: [animals[0], animals[1]], owner: Object.assign({}, user, { pets: [animals[2]] }),
                });
        }
    });
    tester.adapter.createOne.mockImplementation((modelId) => {
        switch (modelId) {
            case "animal":
                return animals[current++];
            case "user":
                return user;
            case "post":
                return post;
        }
    });
    const result = yield tester.query(`mutation M1{
            createPost(input:{createAnimals:[
                    {name:"animal1", birthday:"${date1}"},
                    {name:"animal2"}],
                    createOwner:{name:"user5",
                        createPets:[{name:"pet1"}]
                    }
                } ){
                post{
                    owner{
                        name
                        pets{
                            ...FE
                        }
                    }
                    animals{
                        ...FE
                    }
                }
            }
        }
        fragment FE on AnimalConnection{
            edges{
                node{
                    ...F1
                }
            }
        }
        fragment F1 on Animal{
            id
            name
            age
            birthday
            some
            Weight
            isCat
        }
        `);
    expect(result).toMatchSnapshot();
}));
it("mutationCreate with createOrUpdateSubModel", () => __awaiter(this, void 0, void 0, function* () {
    const tester = new utils_1.default();
    const animal1 = tester.generateModel("animal");
    const user = tester.generateModel("user");
    const post = tester.generateModel("post");
    tester.adapter.findOne.mockImplementation((modelId) => {
        switch (modelId) {
            case "post":
                return Object.assign({}, post, {
                    owner: Object.assign({}, user, {
                        pets: [animal1],
                    }),
                });
        }
    });
    tester.adapter.createOne.mockImplementation((modelId) => {
        switch (modelId) {
            case "animal":
                throw new CreateDuplicateError_1.default("");
            case "user":
                throw new CreateDuplicateError_1.default("");
            case "post":
                return post;
        }
    });
    tester.adapter.updateOne.mockImplementation((modelId) => {
        switch (modelId) {
            case "animal":
                return animal1;
            case "user":
                return user;
            case "post":
                return post;
        }
    });
    tester.adapter.findOrCreateOne.mockImplementation((modelId) => {
        switch (modelId) {
            case "animal":
                return animal1;
            case "user":
                return user;
        }
    });
    const result = yield tester.query(`mutation M1{
        createPost(input:{
            createOrUpdateOwner:{
                create:{
                    name: "OwnerName1"
                    createOrUpdatePets:[{
                        create:{
                            name: "animalNew1"
                        },
                        update: {
                            setName: {name: "updateNameAnimal1"}
                        }
                    }]
                }
                update:{
                    setName:{name: "NewNameOwner"}
                }
            }
        }){
            post{
                owner{
                    pets{
                        edges{
                            node{
                                name
                            }
                        }
                    }
                }
            }
        }
    }`);
    expect(tester.adapter.createOne.mock.calls).toMatchSnapshot();
    expect(tester.adapter.findOrCreateOne.mock.calls).toMatchSnapshot();
    expect(tester.adapter.updateOne.mock.calls).toMatchSnapshot();
    expect(tester.adapter.findOne.mock.calls).toMatchSnapshot();
    expect(result).toMatchSnapshot();
}));
