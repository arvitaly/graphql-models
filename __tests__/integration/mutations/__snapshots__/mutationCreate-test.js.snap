// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`mutation create 1`] = `
Object {
  "createPost": Object {
    "post": Object {
      "animals": Object {
        "edges": Array [
          Object {
            "node": Object {
              "Weight": 1001.1,
              "age": 101,
              "birthday": "Fri, 01 Jan 2010 00:00:00 GMT",
              "id": "QW5pbWFsOjE=",
              "isCat": true,
              "name": "Str1",
              "some": Object {
                "test": 1,
              },
            },
          },
          Object {
            "node": Object {
              "Weight": 1002.1,
              "age": 102,
              "birthday": "Fri, 01 Jan 2010 00:00:00 GMT",
              "id": "QW5pbWFsOjI=",
              "isCat": true,
              "name": "Str2",
              "some": Object {
                "test": 2,
              },
            },
          },
        ],
      },
      "owner": Object {
        "name": "Str1",
        "pets": Object {
          "edges": Array [
            Object {
              "node": Object {
                "Weight": 1003.1,
                "age": 103,
                "birthday": "Fri, 01 Jan 2010 00:00:00 GMT",
                "id": "QW5pbWFsOjM=",
                "isCat": true,
                "name": "Str3",
                "some": Object {
                  "test": 3,
                },
              },
            },
          ],
        },
      },
    },
  },
}
`;

exports[`mutationCreate with createOrUpdateSubModel 1`] = `
Array [
  Array [
    "animal",
    Object {
      "name": "animalNew1",
    },
  ],
  Array [
    "user",
    Object {
      "name": "OwnerName1",
      "pets": Array [
        "1",
      ],
    },
  ],
  Array [
    "post",
    Object {
      "owner": "1",
    },
  ],
]
`;

exports[`mutationCreate with createOrUpdateSubModel 2`] = `
Array [
  Array [
    "animal",
    Object {
      "name": "animalNew1",
    },
  ],
  Array [
    "user",
    Object {
      "createOrUpdatePets": Array [
        Object {
          "create": Object {
            "name": "animalNew1",
          },
          "update": Object {
            "setName": Object {
              "name": "updateNameAnimal1",
            },
          },
        },
      ],
      "name": "OwnerName1",
    },
  ],
]
`;

exports[`mutationCreate with createOrUpdateSubModel 3`] = `
Array [
  Array [
    "animal",
    "1",
    Object {
      "name": "updateNameAnimal1",
    },
  ],
  Array [
    "user",
    "1",
    Object {
      "name": "NewNameOwner",
    },
  ],
]
`;

exports[`mutationCreate with createOrUpdateSubModel 4`] = `
Array [
  Array [
    "post",
    "1",
    Array [
      Object {
        "attribute": Object {
          "model": "user",
          "name": "owner",
          "realName": "owner",
          "required": false,
          "type": "model",
        },
        "fields": Array [
          Object {
            "attribute": Object {
              "model": "animal",
              "name": "pets",
              "realName": "pets",
              "required": false,
              "type": "collection",
            },
            "fields": Array [],
          },
        ],
      },
    ],
  ],
]
`;

exports[`mutationCreate with createOrUpdateSubModel 5`] = `
Object {
  "createPost": Object {
    "post": Object {
      "owner": Object {
        "pets": Object {
          "edges": Array [
            Object {
              "node": Object {
                "name": "Str1",
              },
            },
          ],
        },
      },
    },
  },
}
`;
